### ISSUE TITLE
[PDO] auto-instrumentation closes the span too soon (?)

### ISSUE THREAD

**Issue by dkarlovi** [open-telemetry/opentelemetry-php#1285]
**Describe your environment** 

```
PHP 8.3.6 (cli) (built: Apr 11 2024 15:12:15) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.3.6, Copyright (c) Zend Technologies
    with Zend OPcache v8.3.6, Copyright (c), by Zend Technologies
```

```
opentelemetry hooks => enabled
extension version => 1.0.1
```

```
open-telemetry/api                        1.0.3       API for OpenTelemetry PHP.
open-telemetry/context                    1.0.2       Context implementation for OpenTelemetry PHP.
open-telemetry/exporter-otlp              1.0.4       OTLP exporter for OpenTelemetry.
open-telemetry/gen-otlp-protobuf          1.1.0       PHP protobuf files for communication with OpenTelemetry OTLP collectors/servers.
open-telemetry/opentelemetry-auto-pdo     0.0.12      OpenTelemetry auto-instrumentation for PDO
open-telemetry/opentelemetry-auto-symfony 1.0.0beta23 OpenTelemetry auto-instrumentation for Symfony
open-telemetry/sdk                        1.0.8       SDK for OpenTelemetry PHP.
open-telemetry/sem-conv                   1.24.0      Semantic conventions for OpenTelemetry PHP.
open-telemetry/transport-grpc             1.0.0       gRPC transport for OpenTelemetry PHP.
```

**Steps to reproduce**

1. enable PDO auto-instrumentation
2. do some queries with Symfony+Doctrine
3. export to Jaeger UI

**What is the expected behavior?**

Queries have a duration of about 2ms, according to [Symfony Doctrine Profiler](https://github.com/doctrine/DoctrineBundle/blob/2.12.x/src/DataCollector/DoctrineDataCollector.php).

**What is the actual behavior?**

![Jaeger-UI](https://github.com/open-telemetry/opentelemetry-php/assets/209225/7a11dd71-8017-4259-a1ec-a0e67ce43497)

**Additional context**

It seems the span is closed too soon?

**Comment by ChrisLightfootWild** [comment]
Hey @dkarlovi 👋

As I understand the above trace, there's possibly a gap between the prepare/execute spans. I'm not familiar with the Doctrine profiler, but wonder if comparing the start time of the prepare span and end time of the execute span looks a little closer to 2ms?

Could you please trace `SELECT SLEEP(2)` and see if that still yields similar results on the execute span?

**Comment by dkarlovi** [comment]
@ChrisLightfootWild 

> Could you please trace SELECT SLEEP(2) and see if that still yields similar results on the execute span?

Yes, I'll do that ASAP, thanks for the idea!

**Comment by dkarlovi** [comment]
@brettmc I can see your PR is closing this, am I to assume you've reproduced the issue in some way?

**Comment by brettmc** [comment]
> am I to assume you've reproduced the issue in some way?

I have not. I think I've assumed, based on Chris' comment above, that the profiler is measuring the combined prepare+execute, and that linking them would make it more clear that that's what's happening.

I'm confident that the extension+package are correctly measuring the execution time of the functions that they are wrapping. I guess the info we need from you is "what is the doctrine profiler measuring"? Is anything async happening when it fetches the results, and the profiler stops measuring when the last row is fetched, for instance?

**Comment by dkarlovi** [comment]
I think Doctrine measures the entire "prepare, execute, fetch" as a single block, I didn't get around to testing the suggestion, but will do it tomorrow so we have more info to work with.

**Comment by dkarlovi** [comment]
OK, just checked, [Symfony will indeed measure prepare, execute, fetch together](https://github.com/symfony/doctrine-bridge/blob/5594233de722a7ddc8bdfeeee0cacdbdfb501957/Middleware/Debug/Connection.php#L47-L60). Since we're measuring PDO here, it makes sense they'd differ since the rest of it is Symfony dev logging (not meant for production) + Doctrine overhead.

I've used the suggested `SELECT SLEEP(2)` statement and it indeed takes 2s in `PDOStatement::execute` span.

This means this issue is invalid, but it does raise a question should Doctrine get instrumented separately from PDO because you do want to measure the queries including the overhead.

Closing, thank you all.
